### DISCUSSION TITLE
A functional deploy for Fastlane

### DISCUSSION THREAD

**Discussion by davidakoontz** [skiptools/skip#226] [Ideas]
I really think this is a Fastlane short coming... but since Skip.tools is promoting Fastlane as a dual platform solution.  

**skip init --fastlane --appid=com.goodcompany.HelloWorld hello world HelloWorld**

A complete and functioning Fastlane configuration would go a long way to making my life less painful and happier.  I've spent the last few weeks trying to learn/grok the Skiip/Fastlane integration and much of that time could have been greatly accelerated if Skip would provide a working Fastlane set of configuration files.  Such that a Fastlane deploy would generate 2 working Apps (iOS & Android) and deploy to MY AppStore & PlayStore the HelloWorld starter project.

Think about it...  out of the box HelloWorld is built / certified-signed / deployed / in the App Store.

There are lots of configuration files that I bet SKIP could deliver with INIT.  I'm learning... but it is a very time consuming and error prone path... some good defaults, comments, generic examples would help.

**Comment by davidakoontz** [comment]
A really smart colleague of mine told me:  "Now, is the time to deploy to the Store, the next best time is with HelloWorld."

I would be happy to help with this endeavor if one of the Skippers needed/wanted help.

**Comment by marcprux** [comment]
While it would be wonderful to output a completely configured fastlane outline for a new project, it simply isn't possible. We don't know any of the metadata about your project (name, description, content ratings, etc.), nor can we create your screenshots. And app signing is an intrinsically finicky setup process. The best `skip init` can do is to give you the scaffolding for a Fastlane setup, and leave you to fill in the blanks.

Fastlane is a large and complex topic, and it may not be worth the additional learning for every project. But once you get the configuration working, it can really speed up deploying new versions of your app. I might recommend that you start off with manual deployment to the app stores, and then once you want to establish a release cadence, go through the additional effort of getting fastlane automation working for your project.
